fabric-extensibility-api-v1/src/main/java/net/fabricmc/fabric/mixin/extensibility/AbstractSkeletonEntityMixin.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.extensibility;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.entity.mob.AbstractSkeletonEntity;
+import net.minecraft.item.BowItem;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+
+@Mixin(AbstractSkeletonEntity.class)
+public class AbstractSkeletonEntityMixin {
+	@Redirect(method = "updateAttackType", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;getItem()Lnet/minecraft/item/Item;"))
+	private Item updateAttackType(ItemStack itemStack) {
+		return itemStack.getItem() instanceof BowItem ? Items.BOW : itemStack.getItem();
```

Comment:
Not sure if this is a good idea or not, so let the discussion begin, but should we replace the reference to `BowItem` here with `FabricBow` (or whatever we decide to call it) so that classes don't have to inherit directly from BowItem, but can simply call themselves a FabricBow? It might require some expansion of the FabricBow interface but I think it would be a good idea while we're doing all of this.

Reply:
Yes, I had it that way originally, but then i swapped for some reason.